addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/launch-product.test.js

```diff
@@ -0,0 +1,161 @@
+ *  permissions and limitations under the License.
+ */
+
+const { iteratee } = require('lodash');
+const WorkflowPayload = require('@aws-ee/workflow-engine/lib/workflow-payload');
+const LaunchProduct = require('../launch-product/launch-product');
+
+describe('LaunchProduct', ()=> {
+      const meta = {};
+      const input = {};
+    
+      const lp = new LaunchProduct({
+        step: { config: {} },
+        workflowPayload: new WorkflowPayload({ meta, input, workflowInstance: { steps: [] } }),
+      });
+
+    // afterEach(() => {
+    //     const resolvedInputParams = [];
+    //     const datetime = 0;
+    //     const datetime2 = 0;
+    // })
+
+    describe('checkNamespace', () => {
+        // it('Static name should be transformed to start with analysis- and end with number string and be unique between calls', async () => {
+        //     // Build 
+        //     const resolvedInputParams = [{Key: 'Namespace', Value: 'staticname'}];
+        //     const resolvedInputParams2 = [{Key: 'Namespace', Value: 'staticname'}];
+        //     const datetime = Date.now();
+        //     const datetime2 = Date.now();
+        //     const originalNamespace = resolvedInputParams[0].Value;
+        //     // const originalNamespace2 = resolvedInputParams2[0].Value;
+    
+        //     // Operate
+        //     const { afterNamespace, index } = await lp.getNamespaceAndIndexIfNecessary(resolvedInputParams, datetime);
+        //     const { afterNamespace2, index2 } = await lp.getNamespaceAndIndexIfNecessary(resolvedInputParams2, datetime2);
+    
+        //     // Check
+        //     await expect(afterNamespace.startsWith('analysis-')).toBeTruthy();
+        //     await expect(afterNamespace).not.toBe(afterNamespace2);
+        //     await expect(afterNamespace).toBe('analysis-' + originalNamespace + `-${datetime}`);
+        // });
+
+        it('Static name should be transformed to start with analysis- and end with number string and be unique between calls', async () => {
+            // Build 
+            const resolvedInputParams = [{Key: 'Namespace', Value: 'staticname'}];
+            const datetime = Date.now();
+            const resolvedInputParams2 = [{Key: 'Namespace', Value: 'staticname'}];
+            const datetime2 = Date.now() + 1;
+            const originalNamespace = resolvedInputParams[0].Value;
+            // const originalNamespace2 = resolvedInputParams2[0].Value;
+    
+            // Operate
+            const after = await lp.getNamespaceAndIndexIfNecessary(resolvedInputParams, datetime);
+            const after2 = await lp.getNamespaceAndIndexIfNecessary(resolvedInputParams2, datetime2);
+    
+            // Check
+            const afterNamespace = after[0];
+            const afterNamespace2 = after2[0];
+            await expect(afterNamespace.startsWith('analysis-')).toBeTruthy();
+            await expect(afterNamespace).toBe
+            await expect(afterNamespace).not.toBe(afterNamespace2);
```

Comment:
Aren't these checks redundant? Given that you added the last one. I think other tests have similar redundant checks.

Reply:
Ah, they are. Thank you for pointing that out.